### PR TITLE
feat(RHTAPREL-358): remove releasestrategy field

### PR DESCRIPTION
**Pull request by davidmogar**
This change removes the releasestrategy field from collect-data and from pipelines using this tasks. In addition, extraData is not collected anymore and data is collected instead. This change is triggered by a release-service PR containing both of those changes.

### CODE REVIEW

**Review by mmalina — DISMISSED** [review]

**Comment by davidmogar** [comment]
This change has to be on hold until the release-service change is merged.

File: pipelines/deploy-release/README.md
Comment (johnbieren) [inline pipelines/deploy-release/README.md:19]: This doesn't seem backwards compatible. Per https://github.com/redhat-appstudio/release-service-catalog/pull/212#discussion_r1336968507 shouldn't this be 1.0.0?

Comment (davidmogar) [reply pipelines/deploy-release/README.md:19]: Wondering the same. Removing a parameter shouldn't break any pipeline passing it, isn't it? But the results... maybe.

Comment (mmalina) [reply pipelines/deploy-release/README.md:19]: Yeah, I think Johnny is right. It does seem a bit weird, given what we've been doing until now. But if we want to use semver going forward, we should probably increase the major version here.

File: tasks/collect-data/README.md
Comment (johnbieren) [inline tasks/collect-data/README.md:28]: typo

File: tasks/collect-data/tests/test-collect-data.yaml
Comment (johnbieren) [inline tasks/collect-data/tests/test-collect-data.yaml:57]: these are not the right params for a bundle resolver, they are for a cluster resolver

File: tasks/collect-data/tests/test-collect-data-with-extra-data.yaml
Comment (johnbieren) [inline tasks/collect-data/tests/test-collect-data-with-extra-data.yaml:66]: this is a cluster resolver

File: tasks/collect-data/tests/test-collect-data-with-extra-data.yaml
Comment (johnbieren) [inline tasks/collect-data/tests/test-collect-data-with-extra-data.yaml:100]: It won't let me comment on it, but new lines 105-106 and the check-result task aren't right. They still refer to extraData instead of data

File: tasks/collect-data/tests/test-collect-data-fail-missing-cr.yaml
Comment (johnbieren) [inline tasks/collect-data/tests/test-collect-data-fail-missing-cr.yaml:58]: cluster

File: tasks/collect-data/collect-data.yaml
Comment (johnbieren) [inline tasks/collect-data/collect-data.yaml:93]: is this not becoming data too? Or maybe just not yet because it isn't in the release-service pr?

File: tasks/collect-data/README.md
Comment (johnbieren) [inline tasks/collect-data/README.md:27]: Remove releasestrategy param and result from the task

File: tasks/collect-data/README.md
Comment (johnbieren) [reply tasks/collect-data/README.md:28]: Can you be more descriptive here? Like that we are grabbing a different field in the spec due to API changes and the task result has changed

**Review by johnbieren — CHANGES_REQUESTED** [review]
Comments in line, but the stuff about versioning is relevant throughout

File: tasks/collect-data/collect-data.yaml
Comment (davidmogar) [reply tasks/collect-data/collect-data.yaml:93]: Yep. Fixed.

File: tasks/collect-data/tests/test-collect-data-fail-missing-cr.yaml
Comment (davidmogar) [reply tasks/collect-data/tests/test-collect-data-fail-missing-cr.yaml:58]: Moved to cluster.

File: tasks/collect-data/tests/test-collect-data-with-extra-data.yaml
Comment (davidmogar) [reply tasks/collect-data/tests/test-collect-data-with-extra-data.yaml]: Moved to cluster.

File: tasks/collect-data/tests/test-collect-data.yaml
Comment (davidmogar) [reply tasks/collect-data/tests/test-collect-data.yaml:57]: Moved to cluster.

File: tasks/collect-data/README.md
Comment (davidmogar) [reply tasks/collect-data/README.md]: Done.

**Review by johnbieren — APPROVED** [review]

**Comment by davidmogar** [comment]
Merging with broken tests as we have a cycle dependency (it depends on the release-service API change which also depends on this tasks through E2E). @johnbieren agreed on this as well.

**Pull request merged**
